api/pkg/s3/datastore/aws/aws.go

```diff
@@ -165,6 +165,26 @@ func (ad *AwsAdapter) DELETE(object *pb.DeleteObjectInput, ctx context.Context)
 	return NoError
 }
 
+func (ad *AwsAdapter) RestoreObject(object *pb.RestoreObjectInput, ctx context.Context) S3Error {
+
+	bucket := ad.backend.BucketName
+
+	newObjectKey := object.Bucket + "/" + object.Key
+
+	restoreInput := awss3.RestoreObjectInput{Bucket: &bucket, Key: &newObjectKey}
```

Comment:
if we see the AWS documentation, this is the full RestoreObjectInput. Why are we not using the RestoreRequest parameter ?
`input := &s3.RestoreObjectInput{
    Bucket: aws.String("examplebucket"),
    Key:    aws.String("archivedobjectkey"),
    RestoreRequest: &s3.RestoreRequest{
        Days: aws.Int64(1),
        GlacierJobParameters: &s3.GlacierJobParameters{
            Tier: aws.String("Expedited"),
        },
    },
}`